### PR TITLE
Build for new archs

### PR DESCRIPTION
**Pull request by carterbox**
Refactor the build scripts to target consistent CUDA archs across toolkit versions and enable POWERPC builds.

CUDA compatability is increased. The binaries will now run on all architectures supported by each toolkit. Before, the minimum targeted arch for ppc64le and aarch64 was 50; now it is 35.

Compile-time optimization has decreased. I no longer target 61, these devices can run 60. I tried to target the lowest arch from each named generation. 35 now only gets PTX, so it has to be compiled by the end users' CUDA driver for those devices; this will increase first startup times for those devices. Actually, for Windows, all archs have been getting PTX only because it takes to long to generate the machine code here on the feedstock.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by carterbox** [comment]
I have tried appending a new CUDAToolkit_ROOT to CMAKE_ARGS. IF that doesn't work, try replacing the existing one, so it isn't duplicated.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -98,12 +98,12 @@ source:
...
 
 build:
-  number: 5
+  number: 6
   skip: true  # [cuda_compiler_version == "None"]
   skip: true  # [cuda_compiler_version == "10.2"]
   skip: true  # [cuda_compiler_version == "11.0"]
   skip: true  # [cuda_compiler_version == "11.1"]
-  skip: true  # [ppc64le]
+  skip: true  # [cuda_compiler_version == "11.2"]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:106]: sorry why is this necessary? can we not be compatible with both using little effort?

Comment (carterbox) [reply recipe/meta.yaml:106]: How much longer do you want me to build for 11.2? The 11.2 build takes to long, so I will need to reduce the number of target archs.

Comment (hmaarrfk) [reply recipe/meta.yaml:106]: can you install the newer cuda stuff with pytorch or tensorflow from 11.2? if so then that is ok with me.

We just don't have cuda != 11.2 for either of those.

Comment (carterbox) [reply recipe/meta.yaml:106]: For clarity, the addition of ppc64le is unrelated to the removal of 11.2.

I think you would need an 11.2 build of magma to build pytorch with 11.2. I don't think you can use 11.8 build of magma to build pytorch with 11.2, so I'll try to get the 11.2 build passing again.

Comment (hmaarrfk) [reply recipe/meta.yaml:106]: we could go the other way too, and try to build pytorch and ensorflow with 11.8

**Comment by carterbox** [comment]
The variablility in build times is frustrating; it's like variable by like 15%!

**Comment by hmaarrfk** [comment]
so did you end up "reducing" anything? who will be affected if you did?

**Comment by carterbox** [comment]
Compatability is actually increased. The binaries will run on all architectures supported by the toolkit. Before, the minimum arch for ppc64le or aarch64 was 50; now it is 35.

Optimization has decreased. I no longer target 61 (these devices can run 60), and 35 only gets PTX, so it has to be compiled by the end users' CUDA driver for those devices. Actually, for Windows, all archs have been getting PTX only because it takes to long to generate the machine code.

**Comment by hmaarrfk** [comment]
Does Nvidia provide a table for what "61" means in terms of the products that you buy?

GTX????
RTX????

**Comment by carterbox** [comment]
https://en.wikipedia.org/wiki/CUDA#GPUs_supported

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml]: In term of migrating to CUDA 11.8 both PyTorch ( https://github.com/conda-forge/pytorch-cpu-feedstock/pull/195 ) and TensorFlow ( https://github.com/conda-forge/tensorflow-feedstock/pull/344 ) have updated

Regarding CUDA 11.2, PyTorch dropped it ( https://github.com/conda-forge/pytorch-cpu-feedstock/pull/195 ) and TensorFlow is planning to ( https://github.com/conda-forge/tensorflow-feedstock/pull/347#issuecomment-1805196832 )

So maybe we can simplify the builds here

Comment (carterbox) [reply recipe/meta.yaml]: Thanks! I'll keep that in mind for the next release!

Comment (jakirkham) [reply recipe/meta.yaml]: Note: Tracking in issue ( https://github.com/conda-forge/libmagma-feedstock/issues/12 )
